tests/base.py

```diff
@@ -426,3 +456,74 @@ def create_test_data(self, testable_streams, start_date, start_date_2=None, min_
             self.modify_expected_records(expected_records[stream])
 
         return expected_records
+
+    @staticmethod
+    def _shift_to_start_of_list(key, values):
+        new_list = values.copy()
+
+        if key in values:
+            new_list.remove(key)
+            new_list.insert(0, key)
+
+        return new_list
+
+    def run_initial_sync(self, environment, data_type):
+        # Instantiate connection with default start/end dates
+        conn_id = connections.ensure_connection(self)
+
+        # run in check mode
+        check_job_name = runner.run_check_mode(self, conn_id)
+
+        # verify check exit codes
+        exit_status = menagerie.get_exit_status(conn_id, check_job_name)
+        menagerie.verify_check_exit_status(self, exit_status, check_job_name)
+
+        found_catalogs = menagerie.get_catalogs(conn_id)
+        self.assertGreater(len(found_catalogs), 0, msg="unable to locate schemas for connection {}".format(conn_id))
+
+        found_catalog_names = set(map(lambda c: c['tap_stream_id'], found_catalogs))
+        diff = self.expected_check_streams().symmetric_difference(found_catalog_names)
+        self.assertEqual(len(diff), 0, msg="discovered schemas do not match: {}".format(diff))
+        print("discovered schemas are OK")
+
+        # Select all available fields from all testable streams
+        exclude_streams = self.expected_streams().difference(self.testable_streams(environment, data_type))
+        self.select_all_streams_and_fields(
+            conn_id=conn_id, catalogs=found_catalogs, select_all_fields=True, exclude_streams=exclude_streams
+        )
+
+        catalogs = menagerie.get_catalogs(conn_id)
+
+        # Ensure our selection worked
+        for cat in catalogs:
+            catalog_entry = menagerie.get_annotated_schema(conn_id, cat['stream_id'])
+            # Verify all testable streams are selected
+            selected = catalog_entry.get('annotated-schema').get('selected')
+            print("Validating selection on {}: {}".format(cat['stream_name'], selected))
+            if cat['stream_name'] not in self.testable_streams(environment, data_type):
+                self.assertFalse(selected, msg="Stream selected, but not testable.")
```

Comment:
I don't love that we now have all these assertions in base. Typically we don't want to abstract away what we are testing from the test itself.